src/gpu/nvidia/cudnn_binary.hpp

```diff
@@ -103,7 +103,10 @@ struct cudnn_binary_t : public primitive_t {
                 case f32:
                     return inputs_same
                             && (input_type == f32 || input_type == s8
-                                    || input_type == f16);
+                                    || input_type == f16 || input_type == bf16);
+                case bf16:
+                    return inputs_same
+                            && (input_type == f32 || input_type == bf16);
```

Comment:
I guess bf16 should have some hardware support guard since not all cards can run it...

Reply:
I added a function that check if BF16 is supported for the given device. Please use it in the future PRs.